[Dockerfiles/manifests] Update based on chart version 3.33.6

davidor opened the pull request:
### What does this PR do?

Updates the sample Kubernetes manifests.
Now there are based on the version 3.33.6 of the chart (latest available).

I ran `Dockerfiles/manifests/generate.sh` to update the manifests.


### Describe how to test/QA your changes

Skip.

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

davidor committed:
[Dockerfiles/manifests] Update based on chart version 3.33.6

clamoriniere commented on Dockerfiles/manifests/agent-only/daemonset.yaml:
could you check why this file was removed?

davidor replied:
Sorry, we needed to update the `generate.sh` script to add API keys.
It's needed now because it no longer has a default value in the chart and the agent daemonset is not generated without one.
Should be fixed now.

davidor committed:
[Dockerfiles/manifests/generate.sh] Set API key

Needed now because it no longer has a default value in the chart and the agent
daemonset is not generated without one.

clamoriniere commented on Dockerfiles/manifests/agent-only/README.md:
````diff
@@ -1,9 +1,10 @@
 
 ```yaml
 datadog:
+  apiKey: <DATADOG_API_KEY>
````
should we also add `appKey` ?

davidor replied:
I don't think so, because unlike apiKey, appKey does not appear in an `if` that decides whether a manifest should be generated.

clamoriniere reviewed (approved):
small comment about "appKey:

clamoriniere commented on Dockerfiles/manifests/generate.sh:
```diff
@@ -131,7 +131,7 @@ for values in "$TMPDIR"/values-*.yaml; do
     mkdir "${type:?}"
 
-    helm template --kube-version 1.21 --namespace default datadog "${HELM_DATADOG_CHART:-datadog/datadog}" --values "$values" --output-dir "$TMPDIR/generated_$type"
+    helm template --kube-version 1.21 --namespace default datadog "${HELM_DATADOG_CHART:-datadog/datadog}" --set datadog.apiKey=DUMMY_API_KEY --values "$values" --output-dir "$TMPDIR/generated_$type"
```
last request can be move to kube-version 1.24 ?

davidor replied:
Done :+1:

davidor committed:
[Dockerfiles/manifests/generate.sh] Update to kubernetes 1.24

davidor committed:
[Dockerfiles/manifests] Regenerate for kubernetes 1.24
